ROX-14549: Explaining Red Hat SSO configuration in ACS Console (#56937)
openshift/openshift-docs: operating/manage-user-access/understanding-authentication-providers.adoc

:_content-type: ASSEMBLY
[id="understanding-authentication-providers"]
= Understanding authentication providers
include::modules/common-attributes.adoc[]
:context: authentication-provider-structure

toc::[]

[role="_abstract"]

An authentication provider connects to a third-party source of user identity(for example, an identity provider or IDP) gets the user identity, issues a token based on that identity, and returns the token to RHACS.
This token allows {product-title-short} to authorize the user. 
{product-title-short} uses the token within the user interface and API calls.

After installing RHACS, you must set up your IDP to authorize users.

In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own idP immediately after provisioning.

> [NOTE]
> ====
> If you are using OpenID Connect (OIDC) as your IDP, {product-title-short} relies on mapping rules that examine the values of specific claims like `groups`, `email`, `userid` and `name` from either the user ID token or the `UserInfo` endpoint response to authorize the users.
> If these details are absent, the mapping cannot succeed and the user does not get access to the required resources.
> Therefore, you need to ensure that the claims required to authorize users from your IDP, for example, `groups`, are included in the authentication response of your IDP to enable successful mapping.
> ====
> .Example
> Connect an Azure Active Directory (AD) to {product-title-short} using SSO configuration. For more information, see xref:configuring-identity-providers/connecting-azure-ad-to-rhacs-using-sso-configuration.adoc[Connecting Azure AD to {product-title-short} using SSO configuration]

Reviewer: ```suggestion
[NOTE]
====
If you are using OpenID Connect (OIDC) as your IDP, {product-title-short} relies on mapping rules that examine the values of specific claims like `groups`, `email`, `userid` and `name` from either the user ID token or the `UserInfo` endpoint response to authorize the users.
If these details are absent, the mapping cannot succeed and the user does not get access to the required resources.
Therefore, you need to ensure that the claims required to authorize users from your IDP, for example, `groups`, are included in the authentication response of your IDP to enable successful mapping.
====

.Example

Connect an Azure Active Directory (AD) to {product-title-short} using SSO configuration. For more information, see xref:configuring-identity-providers/connecting-azure-ad-to-rhacs-using-sso-configuration.adoc[Connecting Azure AD to {product-title-short} using SSO configuration]
```

:_content-type: ASSEMBLY
[id="understanding-authentication-providers"]
= Understanding authentication providers
include::modules/common-attributes.adoc[]
:context: authentication-provider-structure

toc::[]

[role="_abstract"]

> An authentication provider connects to a third-party source of user identity(for example, an identity provider or IDP) gets the user identity, issues a token based on that identity, and returns the token to RHACS.

Reviewer: ```suggestion
An authentication provider connects to a third-party source of user identity (for example, an identity provider or IDP) gets the user identity, issues a token based on that identity, and returns the token to {rh-rhacs-first}.
```

> An authentication provider connects to a third-party source of user identity (for example, an identity provider or IDP) gets the user identity, issues a token based on that identity, and returns the token to {rh-rhacs-first}.

Reviewer 2: ```suggestion
An authentication provider connects to a third-party source of user identity (for example, an identity provider or IDP), gets the user identity, issues a token based on that identity, and returns the token to {rh-rhacs-first}.
```

This token allows {product-title-short} to authorize the user. 
{product-title-short} uses the token within the user interface and API calls.

> After installing RHACS, you must set up your IDP to authorize users.

Reviewer: ```suggestion
After installing {product-title-short}, you must set up your IDP to authorize users.
```

> In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own idP immediately after provisioning.

Reviewer: ```suggestion
In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own IDP immediately after provisioning.
```

> In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own IDP immediately after provisioning.

Reviewer 3: ```suggestion
[role="_additional-resources"]
.Additional resources
* xref:../../operating/manage-user-access/configuring-identity-providers/configure-okta-identity-cloud.adoc#configure-okta-identity-cloud[Configuring Okta Identity Cloud as a SAML 2.0 identity provider]
* xref:../../operating/manage-user-access/configuring-identity-providers/configure-google-workspace-identity.adoc#configure-google-workspace-identity[Configuring Google Workspace as an OIDC identity provider]
* xref:../../operating/manage-user-access/configuring-identity-providers/configure-ocp-oauth.adoc#configure-ocp-oauth[Configuring OpenShift Container Platform OAuth server as an identity provider]
* xref:../../operating/manage-user-access/configuring-identity-providers/connecting-azure-ad-to-rhacs-using-sso-configuration.adoc#user-authentication[Connecting Azure AD to RHACS using SSO configuration]
```

:_content-type: ASSEMBLY
[id="understanding-authentication-providers"]
= Understanding authentication providers
include::modules/common-attributes.adoc[]
:context: authentication-provider-structure

toc::[]

[role="_abstract"]

> An authentication provider connects to a third-party source of user identity, for example, an identity provider or IDP, gets the user identity, issues a token based on that identity, and returns the token to RHACS. 

Reviewer 2: Generally we don't use parenthesis but i think you need to put (for example, an identity provider or IDP) in parenthesis because you have three verb actions after that modifier to "third party-source of user identity". Or else rewrite the sentence so in this case, I think parentheses are preferable.

This token allows {product-title-short} to authorize the user. 
{product-title-short} uses the token within the user interface and API calls.

After installing RHACS, you must set up your IDP to authorize users. 

> In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own idP immediately after provisioning.

Reviewer 2: ```suggestion
In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own IDP immediately after provisioning.
```

> In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own idP immediately after provisioning.

Reviewer: ```suggestion
In the {product-title-short} documentation, navigate to *Operating -> Managing user access -> Configuring identity providers*, and select the appropriate module to set up your own idP immediately after provisioning.

[NOTE]
====
If you are using an OpenID Connect (OIDC) as your IDP, {product-title-short} relies on mapping rules that examine the values of specific claims like `groups`, `email`, `userid` and `name` from either the user ID token or the `UserInfo` endpoint response to authorize the users.

If these details are absent, the mapping cannot succeed and the user does not get access to the required resources.
Therefore, you need to ensure that the claims required to authorize users from your IDP, for example, `groups`, are included in the authentication response of your IDP to enable successful mapping.
====

.Example

Connect an Azure Active Directory (AD) to {product-title-short} using SSO configuration. For more information, see <provide-xref-link to operating/manage-user-access/configuring-identity-providers/connecting-azure-ad-to-rhacs-using-sso-configuration.adoc>
```

Reviewer: Please add the NOTE and Example. This content was approved by Alex, thx!

[id="auth-provider-claim-mappings"]
== Claim mappings
A claim is the data an identity provider includes about a user inside the token they issue.

> Using claim mappings, you can specify if RHACS should customize the claim attribute it receives from an IDP by adding it as another attribute in the {product-title-short}-issued token.

Reviewer 2: ```suggestion
Using claim mappings, you can specify if {product-title-short} should customize the claim attribute it receives from an IDP by adding it as another attribute in the {product-title-short}-issued token.
```

> For example, you can map from `roles` in the user identity to `groups` in the {product-title-short}-issued token using claim mapping.

Reviewer 2: ```suggestion
For example, you can map from `roles` in the user identity to `groups` in the {product-title-short}-issued token by using claim mapping.
```

> RHACS uses different default claim mappings for every authentication provider.

Reviewer 2: ```suggestion
{product-title-short} uses different default claim mappings for every authentication provider.
```

[id="auth-provider-oidc-default-claim-mappings"]
=== OIDC default claim mappings
The following list provides the default OIDC claim mappings:

* `sub` to `userid`
* `name` to `name`
* `email` to `email`
* `groups` to `groups`

[id="auth-provider-auth0-default-claim-mappings"]
=== Auth0 default claim mappings
The `Auth0` default claim mappings are the same as the OIDC default claim mappings.

[id="auth-provider-saml20-default-claim-mappings"]
=== SAML 2.0 default claim mappings
The following list applies to SAML 2.0 default claim mappings:

* `Subject.NameID` is mapped to `userid`
* every SAML `AttributeStatement.Attribute` from the response gets mapped to its name

[id="auth-provider-google-iap-default-claim-mappings"]
=== Google IAP default claim mappings
The following list provides the Google IAP default claim mappings:

* `sub` to `userid`
* `email` to `email`
* `hd` to `hd`
* `google.access_levels` to `access_levels`

[id="auth-provider-userpki-default-claim-mappings"]

> === User Certificates default claim mappings

Reviewer 2: ```suggestion
=== User certificates default claim mappings
```
